tinymce-single-react-ui/src/Turducken.js

```diff
@@ -0,0 +1,82 @@
+import BlockToolbar from './components/toolbar/BlockToolbar'
+import InlineToolbar from './components/toolbar/InlineToolbar'
+import TinyMCEReact from './components/tinymce/tinymce-react-ui'
+import {blockList, blockType, blockAlign, getTopLevelBlock} from './utils/tag'
+import '../../shared/post-content'
+import 'assets/stylesheets/main.scss'
+
+let blockOpen  = (collapsed) => (collapsed !== null &&  collapsed) // block  if caret
+let inlineOpen = (collapsed) => (collapsed !== null && !collapsed) // inline if range selection
+
+// get tiny node from the container, and the top level block from the caret node
+let tinyNode = (containerNode) => ((containerNode && containerNode.children.length > 0) ? containerNode.children[0] : null)
+let topLevelBlock = (tinyNode, node) => ((tinyNode && node) ? getTopLevelBlock(tinyNode, node) : null)
+
+// Rect at the start of the Range
+let findStartOfRange = (range) => {
+	// make a collapsed range at the start point
+	if (range) {
+		let r = range.cloneRange();
+		r.setEnd(range.startContainer, range.startOffset);
+		return r.getBoundingClientRect();
+	}
+}
+
+// Rect for the Range
+let rangeRect = (range) => {
+	if (range) {
+		return range.getBoundingClientRect();
+	}
+}
+
+export default function Turducken(props) {
+  let store = props.store
+  let collapsed = store.getState().get('collapsed')
+  let range = store.getState().get('range')
+  let node =  store.getState().get('node') // node of caret or ancestor of range
+  let editorRef = store.getState().get('editorRef')
+  let tiny = tinyNode(editorRef)
+  let topBlock = topLevelBlock(tiny, node)
+  let topRect = rangeRect(topBlock)
+
+  return (
+    <div>
+      { topRect && topRect.width &&
+        <div>
+          <div style={ {position: 'absolute', top: topRect.top + 'px',
+                                              width: topRect.width + 'px',
+                                              height: '2px', background: '#ccc' } }></div>
+
+          <div style={ {position: 'absolute', top: topRect.height + topRect.top + 'px', width: topRect.width + 'px', height: '2px', background: '#ccc' } }></div>
+
+          <div style={ {position: 'absolute', top: topRect.top + 'px', width: '2px', height: topRect.height + 'px', background: '#ccc' } }></div>
+
+          <div style={ {position: 'absolute', top: topRect.top + 'px', width: '2px', height: topRect.height + 'px', left: topRect.width + topRect.left + 'px', background: '#ccc' } }></div>
+
+        </div>
+      }
+      <div>
+        <InlineToolbar isOpen={ inlineOpen(collapsed) }
+          rangeRect={ findStartOfRange(range) }
+          pageYOffset={ window.pageYOffset }
+          node={ node }
+          />
+        <BlockToolbar  isOpen={ blockOpen(collapsed) }
+          blockType={ blockType(topBlock) }
+          blockAlign={ blockAlign(topBlock) }
+          blockRect={ rangeRect(topBlock) }
+          />
+      </div>
+      <TinyMCEReact content={window.content}
```

Comment:
A few other points:

- redux is not tied to react (they point out Deku which I'd not seen : http://redux.js.org/docs/basics/UsageWithReact.html)
- we could use other helper libraries like react-redux, rather than redux directly, which abstract things a bit and make the code cleaner to read, or terser, but a bit more opaque to understand (requires learning that specific library)
- redux works like the ClojureScript approach to app state in things like Reagent & Om which I am more familiar with